_data/team/roles/README.md

```diff
@@ -0,0 +1,43 @@
+# Roles
+
+This directory contains YAML files, one per role.
+
+## Editing
+
+Each YAML file should contain the details of one role only.
+Each role is automatically picked up by the template. To add or remove a role, simply add or remove a file. Do not add non-role documents to this directory that can be parsed by Jekyll.
+
+Each file *must* contain the following tags at top level:
+
+* `role`: The name of the role (e.g. Core library maintenance)
+
+Each role can optionally have the following tags:
+* `description`: text describing the role. This can be written in Markdown.
+* `tasks`: a list of strings describing tasks. This is rendered as a list.
+
+Each role can also optionally have lead/member tags, if there are no subroles. If there are subroles, the below are ignored:
+* `current_leads`: a list of names who are current leads
+* `current_members`: a list of names who are current members in the subgroup
+* `historical_leads`: a list of names who were previously leads
+* `historical_members`: a list of names who were previously members in the subgroup
```

Comment:
Yeah I think I could go either way, and it definitely doesn't make too much sense at the moment! When I came up with the structure I thought we'd update the table annually or otherwise as people changed over, and shift the "current" leads/members into the historical column -- but if we head towards just maintaining a "current" column and keeping snapshots of this table around, that's not necessary either. If anyone has strong opinions about this either way they're very welcome!

As a compromise, until we nail down how we want to deal with archival or past lead/subgroups, how about just calling everyone a historical "member" (i.e. unbolding everyone), and removing the `historical_leads` from the documentation?

Reply:
What about cases where we know who the historical leads have been?

Reply:
I guess my take here is - there have been a lot of things lately that have been concentrated on very few individuals and I think we should recognise that. But removing historical leads we're effectively saying "everyone did the same work".

Reply:
One compromise here - assuming the intent is that "leads doesn't mean that they did the work", is that we put year ranges in the historical column. That way it makes it very obvious who has been doing all the work for how long.

Reply:
Mmmh. I was thinking about this table as the current snaphot plus history, so I didn't think about the future and the possibility of sliding roles from current to historical. 

I don't know how difficult it would be, but I think if roles are clearly defined then a Gantt chart will make more sense to track who did what when, going forward. (Similar to what @IAlibay suggests for historical roles, where I don't know if we have good data for something like that).

> What about cases where we know who the historical leads have been?

If there are such cases, than it is clearly OK to have this distinction. But putting everyone as "lead" doesn't give any information, and also mean "everyone did the same work". This looked to be the case in the current table. I'm not saying that this is the correct way forward, just pointing out that there were two classes but only one was filled (I now understand the intent from @lilyminium's comment).

> One compromise here - assuming the intent is that "leads doesn't mean that they did the work", is that we put year ranges in the historical column (next to each name).

This would be ideal (and that's why I think a Gantt chart makes most sense going forward), but is this information easily available?

Reply:
> I was thinking about this table as the current snaphot plus history,

The current table is not the current snapshot unfortunately - if we're being honest with ourselves, a lot of this is "these folks _will_ take on these roles", not "these folks are currently taking on these roles".

Reply:
Just to make it clear - this entire thing is a catch-22, and it's a set of discussions we as coredevs should have had over several meetings.

Unfortunately we (rather unfairly), just dumped these details on @lilyminium and @fiona-naughton in the hope that it'll get done with an arbitrary deadline.

Reply:
> This would be ideal (and that's why I think a Gantt chart makes most sense going forward), but is this information easily available?

My current understanding is that we largely *don't* have the data to specifically point out who did what when in the 2020-2023 era, and that also this organisational structure did not yet exist -- so that everyone in the "historical contributors" column is a major contributor, who in hindsight could theoretically be described to have done leadery roles. However, unless we start adding and crediting people who did less substantial work, or start drawing thresholds and declaring what amount of work qualifies as being considered a retrospective lead vs member, we don't have a proper distinction between historical "leads" and "members". Currently in both the README and the page I've described the "historical" column as (somewhat vaguely) major contributors or leaders:

**Site page:**
> In addition, we list past leading figures from 2020 onwards in the "Historical contributors" column.

**README for yaml:**
> * `historical_members`: a list of names who previously contributed in a major way

Reply:
> One compromise here - assuming the intent is that "leads doesn't mean that they did the work", is that we put year ranges in the historical column (next to each name). That way it makes it very obvious who has been doing all the work for how long.

Are you suggesting this for the column as it stands now (representing 2020-2023) or for potential future tables when the current lead/member column gets shuffled in or put into a new yearly column? For the former, it seems to me it would be quite difficult to reflect past historical contributions accurately under this organisational framework, especially by the end of week deadline @jennaswa has suggested; I would really be unable to do this myself and need input and re-review from all core devs. For the latter, I would suggest doing an annual column, snapshot/archive, or Gantt chart instead for tidiness.

Reply:
> The current table is not the current snapshot unfortunately - if we're being honest with ourselves, a lot of this is "these folks will take on these roles", not "these folks are currently taking on these roles".

That's true that we haven't directly set when these transitions are meant to happen, thanks for pointing that out. As you said this probably requires core-dev input. Since the next meeting is so far away I'll see if I can get agreement on when the "Current" column starts asynchronously -- if it's either the end of the core-devs role-call or 2024, we can re-title the columns or add explanatory text as appropriate. If it's something else we can perhaps re-visit the idea of dropping the "Current" column in favour of publishing the historical one ASAP, and adding a new column later. Does that feel like a good compromise between getting this table out quickly but accurately to you?

Reply:
> My current understanding is that we largely don't have the data to specifically point out who did what when in the 2020-2023 era, and that also this organisational structure did not yet exist -- so that everyone in the "historical contributors" column is a major contributor, who in hindsight could theoretically be described to have done leadery roles. However, unless we start adding and crediting people who did less substantial work, or start drawing thresholds and declaring what amount of work qualifies as being considered a retrospective lead vs member, we don't have a proper distinction between historical "leads" and "members". 

### Need for clarification

I would make the time frame and intent of "historical" clearer in the table itself - it should be something along the lines of "historical leads 2020-2023". Right now it says "historical contributors" and most folks aren't going to go into the text to check this.

### Dilemmas

#### Maintenance

One of the categories which I'm not sure about (and I don't think the intent of historical had been communicated as "2020-2023 leads" by then) is general maintance:

Here's my dilema:

Over the 2020 to today period, there were a total of 364 PRs against the core library that were labelled under maintainability. The breakdown of PRs submitted under the "maintainability" label for the top 20 submitters is as follows:

![image](https://github.com/MDAnalysis/MDAnalysis.github.io/assets/12460125/ee4b611a-c0b8-4ebf-94c0-7afc036dff17)

Now, it is fair for @jbarnoud to want to be listed under "historical contributors" for that section - @jbarnoud did an overwhelming amount of maintenance in the past (i.e. prior to 2020). However, this raw data indicates that there are other folks here who have, by measure of the number of submitted PRs, have also done a lot of contributions in the 2020 to 2023 period. What is the right way forward here?

#### Releases

Another area where we have strict data on "who did what" is releases.

@richardjgowers was the release manager for releases until 2.0.0. I then did all the releases mostly on lonesome since.

It would make sense here to have that distinction because we have that data on hand?

### Is 2020 too far away? Should we say 2021 instead?

In retrospect, the limit of our data is really 2020, and honestly it's an odd arbitrary limit to put down (I think I may have been at fault here for setting this date for data gathering).

There are things represented in this table that don't really make sense if we say 2020-2023. For example, I was not made core developer until 2021, admittedly I did a lot of work whilst not being a coredev (I think I got to ~ 50 maintenance PRs before folks agreed to make me one...), but I don't think we can claim that I was "managing CI" back then.

May I suggest being a bit more conservative and setting the max retrospective date to 2021?

Reply:
> Are you suggesting this for the column as it stands now (representing 2020-2023) or for potential future tables when the current lead/member column gets shuffled in or put into a new yearly column? 

Both - when we have good data we should do this (see my releases example).

Reply:
> That's true that we haven't directly set when these transitions are meant to happen, thanks for pointing that out. As you said this probably requires core-dev input. Since the next meeting is so far away I'll see if I can get agreement on when the "Current" column starts asynchronously -- if it's either the end of the core-devs role-call or 2024, we can re-title the columns or add explanatory text as appropriate. If it's something else we can perhaps re-visit the idea of dropping the "Current" column in favour of publishing the historical one ASAP, and adding a new column later. Does that feel like a good compromise between getting this table out quickly but accurately to you?

I think I'd be happy with role-call being the set deadline? As far as I see them things are mostly still under the "old model", if we're saying "this will all change by Monday" then I'm looking forward to it.

Reply:
I will say here - I do think that we need to have a clear detailed set of instructions about what it means to be a lead, how we will review this table going forward to ensure that we accurately capture who has been doing what, etc...

I am particularly worried that folks will take on positions right now and another set of unamed folks will end up doing the work and we will never end up acknowledging them.